playback/src/player.rs

```diff
@@ -1305,54 +1305,62 @@ impl PlayerInternal {
                                 // Engineering Society, 60, 399-408.
                                 if self.config.normalisation_method == NormalisationMethod::Dynamic
                                 {
-                                    // steps 1 + 2: half-wave rectification and conversion into dB
-                                    let abs_sample_db = ratio_to_db(sample.abs());
-
-                                    // Some tracks have samples that are precisely 0.0, but ratio_to_db(0.0)
-                                    // returns -inf and gets the peak detector stuck.
-                                    if !abs_sample_db.is_normal() {
-                                        continue;
-                                    }
-
-                                    // step 3: gain computer with soft knee
-                                    let biased_sample = abs_sample_db - threshold_db;
-                                    let limited_sample = if 2.0 * biased_sample < -knee_db {
-                                        abs_sample_db
-                                    } else if 2.0 * biased_sample.abs() <= knee_db {
-                                        abs_sample_db
-                                            - (biased_sample + knee_db / 2.0).powi(2)
-                                                / (2.0 * knee_db)
+                                    // Some tracks have samples that are precisely 0.0. That's silence
+                                    // and we know we don't need to limit that, in which we can spare
+                                    // the CPU cycles.
+                                    //
+                                    // Also, calling `ratio_to_db(0.0)` returns `inf` and would get the
+                                    // peak detector stuck. Also catch the unlikely case where a sample
+                                    // is decoded as `NaN` or some other non-normal value.
+                                    let limiter_db = if sample.is_normal() {
+                                        // step 1-2: half-wave rectification and conversion into dB
+                                        let abs_sample_db = ratio_to_db(sample.abs());
+
+                                        // step 3: gain computer with soft knee
+                                        let bias_db = abs_sample_db - threshold_db;
+                                        let knee_boundary_db = bias_db * 2.0;
+
+                                        let offset_db = if knee_boundary_db < -knee_db {
```

Comment:
Almost there. What this returns isn't an offset, but the absolute sample in dB that's potentially limited. So maybe something like `abs_limited_sample_db`?

Reply:
Sure whatever you'd like,lol!!!

Reply:
There. That var isn't even a thing anymore.